doc/implementation_comparison.md

```diff
@@ -0,0 +1,198 @@
+- features available (see breakdown)
+- maturity level
+- performance (TODO: add data)
+
+## Comparison of supported features 
+
+Beyond the basic RPC functionality, there are a lot of gRPC features that may or may not be supported. The summary
+of supported features in both implementation is available in this section.
+
+### Service config
+
+Service config is currently only supported by Grpc.Core. Right now the feature is not that useful 
+because support for service config encoded in DNS records hasn't been enabled yet by default.
+
+Also see:
+
+- https://github.com/grpc/grpc/blob/master/doc/service_config.md
+- https://github.com/grpc/proposal/blob/master/A2-service-configs-in-dns.md 
+
+### Tracing
+
+grpc-dotnet has tracing bindings that come from ASP.NET Core, and it works with OpenTelemetry SDK, OpenCensus support is TBD.
+Grpc.Core: currently not supported.
+
+### Interceptors
+
+Both implementations support client and server interceptors from Grpc.Core.Interceptors namespace. Interceptors operate at post-deserialization and pre-serialization level (no access to binary payloads).
+
+In addition to gRPC-aware interceptors, grpc-dotnet also allows interception at a HTTP/2 level:
+
+- Incoming gRPC HTTP/2 requests can be processed using [ASP.NET Core middleware](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/middleware/).
+- Outgoing gRPC HTTP/2 requests can be processed using [HttpClient HttpMessageHandlers](https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httpmessagehandler).
+
+### Load Balancing
+
+Grpc.Core provides basic client load balancing policies PICK_FIRST, ROUND_ROBIN and client-lookaside LB policies:
+
+- grpclb - limited use externally as there's no official implementation of the LB policy
+- XDS - Load balancing using the Envoy Universal Data Plane APIs (xDS). Currently this is work in progress but once ready, it will be the LB of choice for lookaside LB.
+
+grpc-dotnet currently doesn't provide any loadbalancing policies. It is likely that to support the XDS loadbalancing policy, features will need to be added to .NET HttpClient.
+
+Proxy loadbalancing is supported by both implementations because loadbalancing is done by a separate process (e.g. Envoy, ngingx etc.) that proxies the traffic.
+
+None of the implementations currently allow user-provided custom loadbalancing policies (= a plugin that provides the loadbalancing logic).
+
+- https://github.com/grpc/grpc/blob/master/doc/load-balancing.md
+- https://github.com/grpc/grpc/blob/master/doc/naming.md
+- https://github.com/grpc/proposal/blob/master/A5-grpclb-in-dns.md
+- https://github.com/grpc/proposal/blob/master/A24-lb-policy-config.md
+
+### Transport support
+
+Both implementations fully support gRPC over HTTP/2 in both TLS and plaintext.
+
+Grpc.Core gets support for other transports supported by C-Core for free. Some (minor) integration work might be required to actually use these transports with gRPC C#.
```

Comment:
What is this section referring to?

Reply:
We don't have tests/examples for some of the transports/features one could potentially use with Grpc.Core and not all of the c-core features are actually exposed in the C# layer, so I added this sentence to not create the impression that all the c-core features will work out of the box.